### PR TITLE
OT275-72: Endpoint Actualizacion de Miembros

### PR DESCRIPTION
**Pull request by Sieglind**
# [OT275-72](https://alkemy-labs.atlassian.net/browse/OT275-72)

Implemented member actualization endpoint at PUT /members/{id}

### HOW HAS THIS BEEN TESTED?

Unit tests and Postman requests were used to test this feature

- [x] Postman.
- [x] Unit test.
- [ ] Integration test.
- [ ] No testing required (only applies for tech task).

**Test Configuration**:

No additional configuration was needed in order to run these tests

**Screenshots**:
Given a member entry in the database:
image
When a PUT request is sent to the endpoint:
image
The results can be seen in the database:
image
When an incorrect body format is sent, validations take place:
image

### CHECKLIST

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/infrastructure/rest/resource/MemberResource.java [hunk 1]
```diff
@@ -71,4 +76,15 @@ public ResponseEntity<GetMemberResponse> add(
     return new ResponseEntity<>(createMemberMapper.toResponse(createdMember), HttpStatus.CREATED);
   }
 
+  @PutMapping(
+      value = "{id}",
+      consumes = MediaType.APPLICATION_JSON_VALUE,
+      produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<GetMemberResponse> update(
+      @PathVariable Long id,
+      @Valid @RequestBody CreateMemberRequest request) {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/resource/MemberResource.java:85]: Al inicio del proyecto, se definió que los request body eran una clase por operación, no se reutiliza la clase. Por lo tanto, aquí debería haber un UpdateMemberRequest.

**Review by mkain0 — CHANGES_REQUESTED** [review]

File: src/main/java/com/alkemy/ong/infrastructure/rest/resource/MemberResource.java (hunk 1 above)
Comment (Sieglind) [reply src/main/java/com/alkemy/ong/infrastructure/rest/resource/MemberResource.java:85]: Aunque tengan los mismos campos?

Comment (mkain0) [reply src/main/java/com/alkemy/ong/infrastructure/rest/resource/MemberResource.java:86]: Si Sr. Lo conversamos allá a lo lejos del inicio de la aceleración cuando les dije que podían reutilizar los DTOs o crear uno por operación y decidieron: option 2

File: src/main/java/com/alkemy/ong/infrastructure/rest/response/member/UpdateMemberResponse.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.alkemy.ong.infrastructure.rest.response.member;
+
+public class UpdateMemberResponse extends GetMemberResponse {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/rest/response/member/UpdateMemberResponse.java:3]: Hace esto es una clara demostración de que no se require la existencia de updateMemberResponse. 

GetMemberResponse es la response del servicio y UpdateMemberRequest la request del servicio aunque tenga los mismos campos y misma validaciones. Ver otros servicios para referencia.

**Review by mkain0 — CHANGES_REQUESTED** [review]

File: src/main/java/com/alkemy/ong/infrastructure/rest/response/member/UpdateMemberResponse.java (hunk 2 above)
Comment (Sieglind) [reply src/main/java/com/alkemy/ong/infrastructure/rest/response/member/UpdateMemberResponse.java:3]: Lei tan mal que termine cambiando cualquier cosa, disculpas

**Review by mkain0 — APPROVED** [review]

**Pull request merged**